RC_1_1 fail to build with Boost-1.70

**Please provide the following information**

libtorrent version (or branch): RC_1_1

platform/architecture: amd64

compiler and compiler version: GCC-8.3

please describe what symptom you see, what you would expect to see instead and
how to reproduce it.
```
In file included from ../include/libtorrent/socks5_stream.hpp:44,
                 from ../include/libtorrent/socket_type.hpp:38,
                 from enum_net.cpp:38:
../include/libtorrent/proxy_base.hpp: In member function ‘libtorrent::io_service& libtorrent::proxy_base::get_io_service()’:
../include/libtorrent/proxy_base.hpp:253:17: error: ‘boost::asio::ip::tcp::socket’ {aka ‘class boost::asio::basic_stream_socket<boost::asio::ip::tcp>’} has no member named ‘get_io_service’
   return m_sock.get_io_service();
                 ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/utp_stream.hpp:37,
                 from ../include/libtorrent/socket_type.hpp:41,
                 from enum_net.cpp:38:
../include/libtorrent/udp_socket.hpp: In member function ‘libtorrent::io_service& libtorrent::udp_socket::get_io_service()’:
../include/libtorrent/udp_socket.hpp:83:53: error: ‘boost::asio::ip::udp::socket’ {aka ‘class boost::asio::basic_datagram_socket<boost::asio::ip::udp>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_ipv4_sock.get_io_service(); }
                                                     ^~~~~~~~~~~~~~
make[1]: *** [Makefile:1036: enum_net.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from ../include/libtorrent/http_stream.hpp:44,
                 from http_stream.cpp:33:
../include/libtorrent/proxy_base.hpp: In member function ‘libtorrent::io_service& libtorrent::proxy_base::get_io_service()’:
../include/libtorrent/proxy_base.hpp:253:17: error: ‘boost::asio::ip::tcp::socket’ {aka ‘class boost::asio::basic_stream_socket<boost::asio::ip::tcp>’} has no member named ‘get_io_service’
   return m_sock.get_io_service();
                 ^~~~~~~~~~~~~~
make[1]: *** [Makefile:1036: http_stream.lo] Error 1
In file included from ../include/libtorrent/tracker_manager.hpp:74,
                 from ../include/libtorrent/torrent.hpp:59,
                 from choker.cpp:37:
../include/libtorrent/udp_socket.hpp: In member function ‘libtorrent::io_service& libtorrent::udp_socket::get_io_service()’:
../include/libtorrent/udp_socket.hpp:83:53: error: ‘boost::asio::ip::udp::socket’ {aka ‘class boost::asio::basic_datagram_socket<boost::asio::ip::udp>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_ipv4_sock.get_io_service(); }
                                                     ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/torrent.hpp:59,
                 from choker.cpp:37:
../include/libtorrent/tracker_manager.hpp: In member function ‘libtorrent::io_service& libtorrent::timeout_handler::get_io_service()’:
../include/libtorrent/tracker_manager.hpp:289:51: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_timeout.get_io_service(); }
                                                   ^~~~~~~~~~~~~~
make[1]: *** [Makefile:1036: choker.lo] Error 1
In file included from ../include/libtorrent/tracker_manager.hpp:74,
                 from ../include/libtorrent/torrent.hpp:59,
                 from ../include/libtorrent/web_connection_base.hpp:63,
                 from web_connection_base.cpp:44:
../include/libtorrent/udp_socket.hpp: In member function ‘libtorrent::io_service& libtorrent::udp_socket::get_io_service()’:
../include/libtorrent/udp_socket.hpp:83:53: error: ‘boost::asio::ip::udp::socket’ {aka ‘class boost::asio::basic_datagram_socket<boost::asio::ip::udp>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_ipv4_sock.get_io_service(); }
                                                     ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/torrent.hpp:59,
                 from ../include/libtorrent/web_connection_base.hpp:63,
                 from web_connection_base.cpp:44:
../include/libtorrent/tracker_manager.hpp: In member function ‘libtorrent::io_service& libtorrent::timeout_handler::get_io_service()’:
../include/libtorrent/tracker_manager.hpp:289:51: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_timeout.get_io_service(); }
                                                   ^~~~~~~~~~~~~~
make[1]: *** [Makefile:1036: web_connection_base.lo] Error 1
In file included from ../include/libtorrent/tracker_manager.hpp:74,
                 from ../include/libtorrent/torrent.hpp:59,
                 from alert.cpp:43:
../include/libtorrent/udp_socket.hpp: In member function ‘libtorrent::io_service& libtorrent::udp_socket::get_io_service()’:
../include/libtorrent/udp_socket.hpp:83:53: error: ‘boost::asio::ip::udp::socket’ {aka ‘class boost::asio::basic_datagram_socket<boost::asio::ip::udp>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_ipv4_sock.get_io_service(); }
                                                     ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/torrent.hpp:59,
                 from alert.cpp:43:
../include/libtorrent/tracker_manager.hpp: In member function ‘libtorrent::io_service& libtorrent::timeout_handler::get_io_service()’:
../include/libtorrent/tracker_manager.hpp:289:51: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_timeout.get_io_service(); }
                                                   ^~~~~~~~~~~~~~
make[1]: *** [Makefile:1036: alert.lo] Error 1
In file included from ../include/libtorrent/tracker_manager.hpp:74,
                 from ../include/libtorrent/torrent.hpp:59,
                 from ../include/libtorrent/bt_peer_connection.hpp:60,
                 from bt_peer_connection.cpp:48:
../include/libtorrent/udp_socket.hpp: In member function ‘libtorrent::io_service& libtorrent::udp_socket::get_io_service()’:
../include/libtorrent/udp_socket.hpp:83:53: error: ‘boost::asio::ip::udp::socket’ {aka ‘class boost::asio::basic_datagram_socket<boost::asio::ip::udp>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_ipv4_sock.get_io_service(); }
                                                     ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/torrent.hpp:59,
                 from ../include/libtorrent/bt_peer_connection.hpp:60,
                 from bt_peer_connection.cpp:48:
../include/libtorrent/tracker_manager.hpp: In member function ‘libtorrent::io_service& libtorrent::timeout_handler::get_io_service()’:
../include/libtorrent/tracker_manager.hpp:289:51: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_timeout.get_io_service(); }
                                                   ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/socks5_stream.hpp:44,
                 from ../include/libtorrent/socket_type.hpp:38,
                 from bt_peer_connection.cpp:65:
../include/libtorrent/proxy_base.hpp: In member function ‘libtorrent::io_service& libtorrent::proxy_base::get_io_service()’:
../include/libtorrent/proxy_base.hpp:253:17: error: ‘boost::asio::ip::tcp::socket’ {aka ‘class boost::asio::basic_stream_socket<boost::asio::ip::tcp>’} has no member named ‘get_io_service’
   return m_sock.get_io_service();
                 ^~~~~~~~~~~~~~
make[1]: *** [Makefile:1036: bt_peer_connection.lo] Error 1
In file included from ../include/libtorrent/tracker_manager.hpp:74,
                 from ../include/libtorrent/aux_/session_impl.hpp:70,
                 from disk_buffer_holder.cpp:34:
../include/libtorrent/udp_socket.hpp: In member function ‘libtorrent::io_service& libtorrent::udp_socket::get_io_service()’:
../include/libtorrent/udp_socket.hpp:83:53: error: ‘boost::asio::ip::udp::socket’ {aka ‘class boost::asio::basic_datagram_socket<boost::asio::ip::udp>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_ipv4_sock.get_io_service(); }
                                                     ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/aux_/session_impl.hpp:70,
                 from disk_buffer_holder.cpp:34:
../include/libtorrent/tracker_manager.hpp: In member function ‘libtorrent::io_service& libtorrent::timeout_handler::get_io_service()’:
../include/libtorrent/tracker_manager.hpp:289:51: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_timeout.get_io_service(); }
                                                   ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/socks5_stream.hpp:44,
                 from ../include/libtorrent/socket_type.hpp:38,
                 from ../include/libtorrent/aux_/session_impl.hpp:80,
                 from disk_buffer_holder.cpp:34:
../include/libtorrent/proxy_base.hpp: In member function ‘libtorrent::io_service& libtorrent::proxy_base::get_io_service()’:
../include/libtorrent/proxy_base.hpp:253:17: error: ‘boost::asio::ip::tcp::socket’ {aka ‘class boost::asio::basic_stream_socket<boost::asio::ip::tcp>’} has no member named ‘get_io_service’
   return m_sock.get_io_service();
                 ^~~~~~~~~~~~~~
make[1]: *** [Makefile:1036: disk_buffer_holder.lo] Error 1
In file included from ../include/libtorrent/socks5_stream.hpp:44,
                 from ../include/libtorrent/socket_type.hpp:38,
                 from ../include/libtorrent/http_connection.hpp:67,
                 from http_connection.cpp:33:
../include/libtorrent/proxy_base.hpp: In member function ‘libtorrent::io_service& libtorrent::proxy_base::get_io_service()’:
../include/libtorrent/proxy_base.hpp:253:17: error: ‘boost::asio::ip::tcp::socket’ {aka ‘class boost::asio::basic_stream_socket<boost::asio::ip::tcp>’} has no member named ‘get_io_service’
   return m_sock.get_io_service();
                 ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/utp_stream.hpp:37,
                 from ../include/libtorrent/socket_type.hpp:41,
                 from ../include/libtorrent/http_connection.hpp:67,
                 from http_connection.cpp:33:
../include/libtorrent/udp_socket.hpp: In member function ‘libtorrent::io_service& libtorrent::udp_socket::get_io_service()’:
../include/libtorrent/udp_socket.hpp:83:53: error: ‘boost::asio::ip::udp::socket’ {aka ‘class boost::asio::basic_datagram_socket<boost::asio::ip::udp>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_ipv4_sock.get_io_service(); }
                                                     ^~~~~~~~~~~~~~
http_connection.cpp: In member function ‘void libtorrent::http_connection::get(const string&, libtorrent::time_duration, int, const libtorrent::aux::proxy_settings*, int, const string&, boost::optional<boost::asio::ip::address>, int, const string&, libtorrent::i2p_connection*)’:
http_connection.cpp:153:11: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   m_timer.get_io_service().post(boost::bind(&http_connection::callback
           ^~~~~~~~~~~~~~
http_connection.cpp:165:11: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   m_timer.get_io_service().post(boost::bind(&http_connection::callback
           ^~~~~~~~~~~~~~
http_connection.cpp: In member function ‘void libtorrent::http_connection::start(const string&, int, libtorrent::time_duration, int, const libtorrent::aux::proxy_settings*, bool, int, boost::optional<boost::asio::ip::address>, int, libtorrent::i2p_connection*)’:
http_connection.cpp:266:11: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   m_timer.get_io_service().post(boost::bind(&http_connection::callback
           ^~~~~~~~~~~~~~
http_connection.cpp:306:13: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
     m_timer.get_io_service().post(boost::bind(&http_connection::callback
             ^~~~~~~~~~~~~~
http_connection.cpp:340:15: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
       m_timer.get_io_service().post(boost::bind(&http_connection::callback
               ^~~~~~~~~~~~~~
http_connection.cpp:352:34: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   instantiate_connection(m_timer.get_io_service()
                                  ^~~~~~~~~~~~~~
http_connection.cpp:361:13: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
     m_timer.get_io_service().post(boost::bind(&http_connection::callback
             ^~~~~~~~~~~~~~
http_connection.cpp:370:12: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
    m_timer.get_io_service().post(boost::bind(&http_connection::callback
            ^~~~~~~~~~~~~~
make[1]: *** [Makefile:1036: http_connection.lo] Error 1
In file included from ../include/libtorrent/tracker_manager.hpp:74,
                 from ../include/libtorrent/torrent.hpp:59,
                 from ../include/libtorrent/web_connection_base.hpp:63,
                 from ../include/libtorrent/http_seed_connection.hpp:55,
                 from http_seed_connection.cpp:43:
../include/libtorrent/udp_socket.hpp: In member function ‘libtorrent::io_service& libtorrent::udp_socket::get_io_service()’:
../include/libtorrent/udp_socket.hpp:83:53: error: ‘boost::asio::ip::udp::socket’ {aka ‘class boost::asio::basic_datagram_socket<boost::asio::ip::udp>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_ipv4_sock.get_io_service(); }
                                                     ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/torrent.hpp:59,
                 from ../include/libtorrent/web_connection_base.hpp:63,
                 from ../include/libtorrent/http_seed_connection.hpp:55,
                 from http_seed_connection.cpp:43:
../include/libtorrent/tracker_manager.hpp: In member function ‘libtorrent::io_service& libtorrent::timeout_handler::get_io_service()’:
../include/libtorrent/tracker_manager.hpp:289:51: error: ‘libtorrent::deadline_timer’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>’} has no member named ‘get_io_service’
   io_service& get_io_service() { return m_timeout.get_io_service(); }
                                                   ^~~~~~~~~~~~~~
In file included from ../include/libtorrent/socks5_stream.hpp:44,
                 from ../include/libtorrent/socket_type.hpp:38,
                 from ../include/libtorrent/aux_/session_impl.hpp:80,
                 from http_seed_connection.cpp:52:
../include/libtorrent/proxy_base.hpp: In member function ‘libtorrent::io_service& libtorrent::proxy_base::get_io_service()’:
../include/libtorrent/proxy_base.hpp:253:17: error: ‘boost::asio::ip::tcp::socket’ {aka ‘class boost::asio::basic_stream_socket<boost::asio::ip::tcp>’} has no member named ‘get_io_service’
   return m_sock.get_io_service();
                 ^~~~~~~~~~~~~~
make[1]: *** [Makefile:1036: http_seed_connection.lo] Error 1
make[1]: Leaving directory '/tmp/portage/net-libs/libtorrent-rasterbar-1.1.9999/work/libtorrent-rasterbar-1.1.9999/src'
make: *** [Makefile:625: all-recursive] Error 1
```

--- a/include/libtorrent/io_service.hpp
+++ b/include/libtorrent/io_service.hpp
@@ -47,6 +47,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #endif
 
 #include <boost/asio/io_service.hpp>
+#include <boost/version.hpp>
 
 #if defined TORRENT_BUILD_SIMULATOR
 #include "simulator/simulator.hpp"
@@ -57,6 +58,8 @@ POSSIBILITY OF SUCH DAMAGE.
 #ifdef __OBJC__
 #undef Protocol
 #endif
+
+#include "libtorrent/io_service_fwd.hpp"
 
 namespace libtorrent
 {

--- a/include/libtorrent/io_service_fwd.hpp
+++ b/include/libtorrent/io_service_fwd.hpp
@@ -69,6 +69,14 @@ namespace libtorrent
 #else
 	typedef boost::asio::io_service io_service;
 #endif
+
+#if BOOST_VERSION >= 107000
+template <typename T>
+io_service& get_io_service(T& o) { return static_cast<io_service&>(o.get_executor().context()); }
+#else
+template <typename T>
+io_service& get_io_service(T& o) { return o.get_io_service(); }
+#endif
 }
 
 #endif

--- a/include/libtorrent/proxy_base.hpp
+++ b/include/libtorrent/proxy_base.hpp
@@ -250,7 +250,7 @@ public:
 
 	io_service& get_io_service()
 	{
-		return m_sock.get_io_service();
+		return lt::get_io_service(m_sock);
 	}
 
 	lowest_layer_type& lowest_layer()

--- a/include/libtorrent/tracker_manager.hpp
+++ b/include/libtorrent/tracker_manager.hpp
@@ -286,7 +286,7 @@ namespace libtorrent
 		virtual void on_timeout(error_code const& ec) = 0;
 		virtual ~timeout_handler() {}
 
-		io_service& get_io_service() { return m_timeout.get_io_service(); }
+		io_service& get_io_service() { return lt::get_io_service(m_timeout); }
 
 	private:
 

--- a/include/libtorrent/udp_socket.hpp
+++ b/include/libtorrent/udp_socket.hpp
@@ -80,7 +80,7 @@ namespace libtorrent
 		};
 
 		bool is_open() const { return m_abort == false; }
-		io_service& get_io_service() { return m_ipv4_sock.get_io_service(); }
+		io_service& get_io_service() { return lt::get_io_service(m_ipv4_sock); }
 
 		void subscribe(udp_socket_observer* o);
 		void unsubscribe(udp_socket_observer* o);

--- a/src/http_connection.cpp
+++ b/src/http_connection.cpp
@@ -150,7 +150,7 @@ void http_connection::get(std::string const& url, time_duration timeout, int pri
 
 	if (ec)
 	{
-		m_timer.get_io_service().post(boost::bind(&http_connection::callback
+		lt::get_io_service(m_timer).post(boost::bind(&http_connection::callback
 			, me, ec, static_cast<char*>(NULL), 0));
 		return;
 	}
@@ -162,7 +162,7 @@ void http_connection::get(std::string const& url, time_duration timeout, int pri
 		)
 	{
 		error_code err(errors::unsupported_url_protocol);
-		m_timer.get_io_service().post(boost::bind(&http_connection::callback
+		lt::get_io_service(m_timer).post(boost::bind(&http_connection::callback
 			, me, err, static_cast<char*>(NULL), 0));
 		return;
 	}
@@ -263,7 +263,7 @@ void http_connection::start(std::string const& hostname, int port
 
 	if (ec)
 	{
-		m_timer.get_io_service().post(boost::bind(&http_connection::callback
+		lt::get_io_service(m_timer).post(boost::bind(&http_connection::callback
 			, me, ec, static_cast<char*>(NULL), 0));
 		return;
 	}
@@ -303,7 +303,7 @@ void http_connection::start(std::string const& hostname, int port
 
 			if (i2p_conn->proxy().type != settings_pack::i2p_proxy)
 			{
-				m_timer.get_io_service().post(boost::bind(&http_connection::callback
+				lt::get_io_service(m_timer).post(boost::bind(&http_connection::callback
 					, me, error_code(errors::no_i2p_router), static_cast<char*>(NULL), 0));
 				return;
 			}
@@ -337,7 +337,7 @@ void http_connection::start(std::string const& hostname, int port
 					m_ssl_ctx->set_verify_mode(ssl::context::verify_none, ec);
 					if (ec)
 					{
-						m_timer.get_io_service().post(boost::bind(&http_connection::callback
+						lt::get_io_service(m_timer).post(boost::bind(&http_connection::callback
 								, me, ec, static_cast<char*>(NULL), 0));
 						return;
 					}
@@ -349,7 +349,7 @@ void http_connection::start(std::string const& hostname, int port
 		// assume this is not a tracker connection. Tracker connections that
 		// shouldn't be subject to the proxy should pass in NULL as the proxy
 		// pointer.
-		instantiate_connection(m_timer.get_io_service()
+		instantiate_connection(lt::get_io_service(m_timer)
 			, proxy ? *proxy : null_proxy, m_sock, userdata, NULL, false, false);
 
 		if (m_bind_addr)
@@ -358,7 +358,7 @@ void http_connection::start(std::string const& hostname, int port
 			m_sock.bind(tcp::endpoint(*m_bind_addr, 0), ec);
 			if (ec)
 			{
-				m_timer.get_io_service().post(boost::bind(&http_connection::callback
+				lt::get_io_service(m_timer).post(boost::bind(&http_connection::callback
 					, me, ec, static_cast<char*>(NULL), 0));
 				return;
 			}
@@ -367,7 +367,7 @@ void http_connection::start(std::string const& hostname, int port
 		setup_ssl_hostname(m_sock, hostname, ec);
 		if (ec)
 		{
-			m_timer.get_io_service().post(boost::bind(&http_connection::callback
+			lt::get_io_service(m_timer).post(boost::bind(&http_connection::callback
 				, me, ec, static_cast<char*>(NULL), 0));
 			return;
 		}

--- a/src/lsd.cpp
+++ b/src/lsd.cpp
@@ -112,12 +112,12 @@ void lsd::debug_log(char const* fmt, ...) const
 void lsd::start(error_code& ec)
 {
 	m_socket.open(boost::bind(&lsd::on_announce, self(), _1, _2, _3)
-		, m_broadcast_timer.get_io_service(), ec);
+		, lt::get_io_service(m_broadcast_timer), ec);
 	if (ec) return;
 
 #if TORRENT_USE_IPV6
 	m_socket6.open(boost::bind(&lsd::on_announce, self(), _1, _2, _3)
-		, m_broadcast_timer.get_io_service(), ec);
+		, lt::get_io_service(m_broadcast_timer), ec);
 #endif
 }
 

--- a/src/natpmp.cpp
+++ b/src/natpmp.cpp
@@ -89,7 +89,7 @@ void natpmp::start()
 	mutex::scoped_lock l(m_mutex);
 
 	error_code ec;
-	address gateway = get_default_gateway(m_socket.get_io_service(), ec);
+	address gateway = get_default_gateway(lt::get_io_service(m_socket), ec);
 	if (ec)
 	{
 		char msg[200];

--- a/src/upnp.cpp
+++ b/src/upnp.cpp
@@ -135,7 +135,7 @@ void upnp::start()
 {
 	error_code ec;
 	m_socket.open(boost::bind(&upnp::on_reply, self(), _1, _2, _3)
-		, m_refresh_timer.get_io_service(), ec);
+		, lt::get_io_service(m_refresh_timer), ec);
 
 	m_mappings.reserve(10);
 }

--- a/test/test_fast_extension.cpp
+++ b/test/test_fast_extension.cpp
@@ -453,7 +453,7 @@ boost::shared_ptr<torrent_info> setup_peer(tcp::socket& s, sha1_hash& ih
 	}
 	else
 	{
-		tcp::acceptor l(s.get_io_service());
+		tcp::acceptor l(lt::get_io_service(s));
 		l.open(tcp::v4());
 		l.bind(tcp::endpoint(address_v4::from_string("127.0.0.1")
 			, 3000 + rand() % 60000));

